### ISSUE TITLE
`exec` builtin doesn't take flags

### ISSUE THREAD

**Issue by lilyball** [fish-shell/fish-shell#5779]
The `exec` builtin seems to support parsing flags (based on error messages if you try and provide one), but it doesn't actually define any flags. Bash's `exec` takes 3 flags, 2 of which are clearly useful and one of which is questionable. I think we should at least add support for the `-l` and `-a` flags.

- `-l`: If the first argument to `exec` is `-l` then this prefixes the zeroth arg with a dash as `login` does.
- `-a name`: This flag passes `name` as the zeroth arg.

`exec -l foo` is clearly just shorthand for `exec -a -foo foo`, but is admittedly convenient.

As it is, if I want to replace my fish shell with a command using a custom name I need to write something like `exec bash -c 'exec -a foo bar'` which is silly.

**Comment by floam** [comment]
Yeah, I'd like `-a` or an `--arg0`. As it is, I have to write small C programs to affect these.

**Comment by floam** [comment]
These could also be arguments for `command`.
